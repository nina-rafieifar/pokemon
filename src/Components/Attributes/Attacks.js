import { html, LitElement, css } from "lit-element";
class PokemonAttacks extends LitElement {
  static get properties() {
    return {
      name: { attribute: true },
      damage: { attribute: true }
    };
  }
  static get styles() {
    return css`
      .chip {
        display: inline-block;
        background: #14855a;
        padding: 0 12px;
        border-radius: 5px;
        font-size: 13px;
        font-family: Space Mono;
        display: inline-flex;
        justify-content: space-around;
        min-width: 6rem;
        color: #ffffff;
        width: max-content;
      }
      .pokemon__details_item {
        padding: 0.15rem;
        height: 1.4rem;
      }
    `;
  }
  render() {
    return html`<div class="pokemon__details_item">
      <div class="chip">
        <span>${this.name}</span>
        |
        <span>${this.damage}</span>
      </div>
    </div>`;
  }
}

customElements.define("pokemon-attacks", PokemonAttacks);
