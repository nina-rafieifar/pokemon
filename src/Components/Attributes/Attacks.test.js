import "./Attacks";
import { fixture } from "@open-wc/testing";
import { html, render } from "lit-html";

test("pokemon-attacks renders the correct elements", () => {
  render(html`<pokemon-attacks name="Sludge Bomb" damage="55"></pokemon-attacks>`, document.body);
  expect(document.body.getElementsByTagName("pokemon-attacks")).toBeDefined();
});

test("pokemon-attacks renders as it should", async () => {
  const element = await fixture(html`<pokemon-attacks name="Sludge Bomb" damage="55"></pokemon-attacks>`);
  expect(element).toMatchSnapshot();
});
