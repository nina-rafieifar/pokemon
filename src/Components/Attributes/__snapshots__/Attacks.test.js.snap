// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pokemon-attacks renders as it should 1`] = `
<pokemon-attacks
  damage="55"
  name="Sludge Bomb"
/>
`;
